src/ir_def/construct.ml

```diff
@@ -727,3 +740,16 @@ let objE sort typ_flds flds =
   go [] [] [] flds
 
 let recordE flds = objE T.Object [] flds
+
+let check_call_perform_status success mk_failure =
+  ifE
+    (callE
+      (varE (var "@call_succeeded"
+        T.(Func (Local, Returns, [], [], [bool]))))
+      [] (unitE ()))
+    success
+    (mk_failure
+      (callE
+        (varE (var "@call_error"
+          T.(Func (Local, Returns, [], [], [T.error]))))
```

Comment:
`T.(Func (Local, Returns, [], [], [error]))))`

Reply:
Agreed, fixing in #3727